src/OrleansProviders/Streams/Common/Monitors/DefaultBlockPoolMonitor.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Orleans.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
```

Comment:
Generally it's better to avoid virtual for performance reasons unless it's necessary.  In these classes, the abstraction is provided by the interface, so I don't think we need these to be virtual.